case-server/src/main/java/com/powsybl/caseserver/CaseService.java

```diff
@@ -311,4 +320,42 @@ static void validateCaseName(String caseName) {
             throw CaseException.createIllegalCaseName(caseName);
         }
     }
+
+    List<CaseInfos> searchCases(String query) {
+        checkStorageInitialization();
+        List<DateTime> dates = new ArrayList<>();
+        List<EntsoeGeographicalCode> entsoeCodes = new ArrayList<>();
+
+        String[] searchFields = query.split(" ");
+        for (String searchField : searchFields) {
+            String[] field = searchField.split(":");
+            if (field.length > 1) {
+                if (field[0].equals("date")) {
+                    String date = field[1].trim();
+                    if (!StringUtils.isEmpty(date)) {
+                        dates.add(parseDateTime(date));
+                    }
+                } else if (field[0].equals("tso")) {
+                    String[] tsos = field[1].trim().split("\\|");
```

Comment:
It's either '||' or 'OR' (lucene query syntax)

https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html

> q
>
>    (Optional, string) Query in the Lucene query string syntax.
>
>   You can use the q parameter to run a query parameter search. Query parameter searches do not support the full Elasticsearch Query DSL but are handy for testing.

Reply:
We need parenthesis support (this way, when we have a real elasticsearch, we can directly pass the query to it)
```tso:(foo OR bar)```

To make it simpler, we can make them mandatory for now, even for only one tso, it should work:
```tso:(foo)```

Reply:
So, if i understand well, the query string could be :    date:XXX AND tsos:(X OR Y OR Z)

Is that correct ?

Reply:
done

Reply:
Done